lib/set.rb

```diff
@@ -474,7 +474,15 @@ def -(enum)
   #     Set['a', 'b', 'z'] & ['a', 'b', 'c']    #=> #<Set: {"a", "b"}>
   def &(enum)
     n = self.class.new
-    do_with_enum(enum) { |o| n.add(o) if include?(o) }
+    if enum.is_a?(Set)
```

Comment:
```suggestion
    if enum.is_a?(self.class)
```

Reply:
This suggestion looks like a good idea, but it can do a lot of harm when using inheritance.
Having `Set` as a verification secures most of the cases when having multiple levels of inheritance that will break if using `self.class`